launchpod/angular-launchpod/backend/src/test/java/com/google/launchpod/TranslationServletTest.java

```diff
@@ -0,0 +1,180 @@
+  private static final String TEST_PODCAST_TITLE = "American Life";
+  private static final String TEST_TARGET_LANGUAGE = "es";
+  private static final String TEST_LANGUAGE = "en";
+  private static final String TEST_DESCRIPTION = "Episode generated by launchpod";
+  private static final String TEST_MP3_LINK = "this-is-a-test.mp3";
+  private static final String TEST_ID = "123456";
+  private static final String TEST_EMAIL = "123@abc.com";
+  private static final String BASE_URL = "https://launchpod-step18-2020.appspot.com/rss-feed?id=";
+  private static final RSS TEST_RSS_FEED = new RSS(TEST_PODCAST_TITLE, TEST_DESCRIPTION, TEST_LANGUAGE, TEST_EMAIL,
+      TEST_MP3_LINK);
+  private RemoteTranslateHelper translateHelper = RemoteTranslateHelper
+      .create("AIzaSyBlu9s7xFLjlHAdvlVuUISq_MbyELuCRZo");
+  private Translate translateMock = translateHelper.getOptions().getService();
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Test translate helper to ensure it translates the given string
+   */
+  @Test
+  public void doPost_TranslateHelperWorks() {
+    String sourceLang = translateMock.detect(TEST_DESCRIPTION).getLanguage();
+    Translation translation = translateMock.translate(TEST_DESCRIPTION, TranslateOption.sourceLanguage(sourceLang),
+        TranslateOption.targetLanguage(TEST_TARGET_LANGUAGE));
+
+    String translatedText = translation.getTranslatedText().toLowerCase();
+    assertEquals("episodio generado por launchpod", translatedText);
+  }
+
+  /**
+   * test to ensure that when rss feed is translated the RSS object is modified as
+   * well
+   * 
+   * @throws JsonProcessingException
+   */
+  @Test
+  public void translate_translatesRSSFieldsCorrectly() throws JsonProcessingException {
+    RSS rssTranslatedFeed = TEST_RSS_FEED;
+    // Test channel description translation
+    Translation translation = translateMock.translate(rssTranslatedFeed.getChannel().getDescription(),
+        TranslateOption.targetLanguage(TEST_TARGET_LANGUAGE));
+    rssTranslatedFeed.getChannel().setDescription(translation.getTranslatedText());
+    assertEquals("episodio generado por launchpod", rssTranslatedFeed.getChannel().getDescription().toLowerCase());
+
+    // Test item
+    for (Item item : rssTranslatedFeed.getChannel().getItem()) {
+      translation = translateMock.translate(item.getTitle(), TranslateOption.targetLanguage(TEST_TARGET_LANGUAGE));
+      item.setTitle(translation.getTranslatedText());
+      assertEquals("vida americana", item.getTitle().toLowerCase());
+
+      // set item description
+      translation = translateMock.translate(item.getDescription(),
+          TranslateOption.targetLanguage(TEST_TARGET_LANGUAGE));
+      item.setDescription(translation.getTranslatedText());
+      assertEquals("episodio generado por launchpod", item.getDescription().toLowerCase());
+    }
+  }
```

Comment:
These two tests aren't actually testing the doPost method, which is what we should be testing. It's only testing RemoteTranslateHelper is working correctly. To test doPost, we need to call doPost.

I think the proper way to do it would be to create a new constructer for TranslationServlet(Translation translation) that sets the translation service. For tests we can do TranslationServlet servlet = new TranslationServlet(translateHelper.getOptions().getService())

I'm okay with approving now to unblock you. we can add a todo to fix this